corehq/apps/data_interfaces/models.py

```diff
@@ -1166,7 +1166,8 @@ def _handle_case_duplicate(self, case, rule):
         dedupe_load_counter('unknown', case.domain)()
 
         if not case_matching_rule_criteria_exists_in_es(case, rule):
-            ALLOWED_ES_DELAY = timedelta(hours=1)
+            # refresh interval + avg time to refresh + extra buffer = 1 minute
```

Comment:
I thought that 1 hour is to accomodate for the pillow lag and not the elasticsearch delay?

Reply:
Ah yeah I don't think it was initially set to 1 hour to accommodate that specifically (Matt can correct me if I'm wrong), but it is true that it helps with that issue. Here's how I think about it.

The case to es processor and dedupe processor both run within the case pillow, so the lag we actually care about is between when the case to es processor ran and when the dedupe processor ran. It seems more than likely that when the dedupe processor runs, the updated case will not be available in ES, so we have to resave it. The problem is that we are determining lag based on the server_modified_on property, which pillow lag is going to impact.

To your point, this change is too extreme in its current form because as soon as pillow lag is above 1 minute, given that it seems likely the dedupe processor won't be able to find the case in ES on its first attempt, we won't retry it since enough time has past since the initial server_modified_on time.